From e74cca78262c2ed4a908302ba002ce7367e9602d Mon Sep 17 00:00:00 2001
From: David Gow <david@ingeniumdigital.com>
Date: Mon, 4 Apr 2022 19:01:00 +0800
Subject: [PATCH] Support '7zip' as well as 'p7zip'

7zip 21.0 finally provided an official native port of 7zip, and has superseded
the p7zip port in some distros (openSUSE, for instance, no longer packages
p7zip).  The CLI interface between them is almost identical, but Ark doesn't
recognise the different version string, so doesn't work with 7zip (making it
impossible to open 7z archives in Ark on openSUSE).

Update the 7zip CLI plugin to detect both p7zip and 7zip's version lines.
---
 plugins/cli7zplugin/cliplugin.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/plugins/cli7zplugin/cliplugin.cpp b/plugins/cli7zplugin/cliplugin.cpp
index 1781117a..f1803c76 100644
--- a/plugins/cli7zplugin/cliplugin.cpp
+++ b/plugins/cli7zplugin/cliplugin.cpp
@@ -112,6 +112,7 @@ bool CliPlugin::readListLine(const QString& line)
     }
 
     const QRegularExpression rxVersionLine(QStringLiteral("^p7zip Version ([\\d\\.]+) .*$"));
+    const QRegularExpression rxVersionLine7z(QStringLiteral("^7-Zip \\(z\\) ([\\d\\.]+) .*$"));
     QRegularExpressionMatch matchVersion;
 
     switch (m_parseState) {
@@ -121,6 +122,14 @@ bool CliPlugin::readListLine(const QString& line)
             m_parseState = ParseStateHeader;
             const QString p7zipVersion = matchVersion.captured(1);
             qCDebug(ARK) << "p7zip version" << p7zipVersion << "detected";
+	    break;
+        }
+	matchVersion = rxVersionLine7z.match(line);
+        if (matchVersion.hasMatch()) {
+            m_parseState = ParseStateHeader;
+            const QString l7zipVersion = matchVersion.captured(1);
+            qCDebug(ARK) << "7zip version" << l7zipVersion << "detected";
+	    break;
         }
         break;
 
-- 
2.36.1

